lib/config/config.js

```diff
@@ -0,0 +1,44 @@
+const rcs = require('rcs-core');
+const path = require('path');
+const json = require('json-extra');
+const minimatch = require("minimatch");
+
+class Config {
+  constructor() {
+    this.ignorePatterns = [];
+  }
+
+  load(pathString = path.join(process.cwd(), '.rcsrc')) {
+    let configObject;
+
+    configObject = json.readToObjSync(pathString);
+
+    if (!configObject) {
+      // package.json .rcs if no other config is found
+      configObject = json.readToObjSync(path.join(process.cwd(), 'package.json')).rcs;
+    }
+
+    if (configObject.exclude) {
+      rcs.selectorsLibrary.setExclude(configObject.exclude);
+    }
+
+    if (configObject.reserve) {
+      rcs.selectorsLibrary.setReserved(configObject.reserve);
+    }
+
+    if (configObject.ignore) {
+      this.ignorePatterns = configObject.ignore;
+    }
+  }
+
+  isIgnored(filePath) {
+    for (let i = 0; i < this.ignorePatterns.length; i++) {
+      if (minimatch(filePath, this.ignorePatterns[i])) {
+        return true;
+      }
+    }
+    return false;
+  }
+}; // /include
+
+module.exports = new Config;
```

Comment:
I actually don't know how that works, also I've never seen it. Could you add the parentheses to the class `new Config();`

You know if there are any differences between `new Config` and `new Config()`, just curious.

Reply:
That's valid JS if there is not parameter to the constructor. The linter usually refuse it because it can lead to code like this: `new Date.toString()` which tries to invoke toString as the constructor of some object `Date` so it usually not the developer intention. I'm changing it to be consistent.

Reply:
Ah well, thanks for the answer. That makes sense :)